test/__snapshots__/codegen.test.ts.snap

```diff
@@ -79,6 +79,9 @@ exports[`codegen should generate correct TypeScript 1`] = `
   XataRecord,
 } from \\"@xata.io/client\\";
 
+type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
+type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
```

Comment:
I've testing locally and I think this adds a bit of complexity to the user:

Code in main:

<img width="558" alt="image" src="https://user-images.githubusercontent.com/50486/158773332-2fc61e68-d4fe-41e5-a7ab-859f483846ed.png">

Code in this PR:

<img width="606" alt="image" src="https://user-images.githubusercontent.com/50486/158773359-747bdf29-1071-4c2e-8005-d50a96f20fb5.png">

[We talked](https://github.com/xataio/client-ts/issues/25#issuecomment-1069061003) about just making the `databaseURL` field optional:


```diff
  export type XataClientOptions = {
    fetch?: unknown;
-   databaseURL: string;
+   databaseURL?: string;
    branch: BranchStrategyOption;
    apiKey: string;
    repositoryFactory?: RepositoryFactory;
  };
```

I think that's simpler and there are no side effects. The constructor of `BaseClient` already checks if databaseURL is defined. So that's the only change required and we can get rid of the PartialBy.

I'm happy to make this change directly in this PR if you want.

Reply:
I respectfully disagree because:

1. When comparing the developer experience of main vs. this PR, both work quite well with autocompletion, with the `PartialBy` case making it quite explicit what's missing (`apiKey` and `branch`). Here's a side-by-side comparison, where both cases guide the user by telling them what exactly is missing, and providing autocompletion options:

|Main|This PR|
|---|---|
|![2022-03-17 10 03 23](https://user-images.githubusercontent.com/9947422/158775439-20083d85-9188-4997-bbff-2d73ec393b6d.gif)|![2022-03-17 10 03 47](https://user-images.githubusercontent.com/9947422/158775471-4d60b7c3-d84c-491f-8b55-a180e5f2fb22.gif)|

2. `XataClientOptions` in main is pretty much as opaque as its `PartialBy` counterpart in this PR,

3. We did talk about `databaseURL` being optional, it is now.

For these reasons, I believe there is no extra complexity to the user because they are clearly guided by autocompletion and type hints as in the [GIF above](https://user-images.githubusercontent.com/9947422/158775471-4d60b7c3-d84c-491f-8b55-a180e5f2fb22.gif). Still, if you insist, feel free to make any changes this PR that you believe is best. 🙌